static/app/components/searchQueryBuilder/hooks/useQueryBuilderState.tsx

```diff
@@ -147,51 +148,99 @@ function deleteQueryTokens(
 function modifyFilterOperator(
   query: string,
   token: TokenResult<Token.FILTER>,
-  newOperator: TermOperator
+  newOperator: TermOperatorNew
 ): string {
   if (isDateToken(token)) {
     return modifyFilterOperatorDate(query, token, newOperator);
   }
 
-  const isNotEqual = newOperator === TermOperator.NOT_EQUAL;
-
   const newToken: TokenResult<Token.FILTER> = {...token};
-  newToken.operator = isNotEqual ? TermOperator.DEFAULT : newOperator;
+  const isNotEqual = newOperator === TermOperatorNew.NOT_EQUAL;
+
   newToken.negated = isNotEqual;
+  // if the operator is contains, update the token (output value) to be wrapped in stars
+  if (newOperator === TermOperatorNew.CONTAINS) {
+    newToken.operator = TermOperator.DEFAULT;
+    if (newToken.value.type === Token.VALUE_TEXT) {
+      if (!newToken.value.value.startsWith('*')) {
```

Comment:
This logic seems potentially brittle and probably doesn't address edge cases like when the value is wrapped in quotes. I'd suggest that we create a parser that can identify when a filter is in the format `key:*value*`. It will output an object representation of the filter that you can use to add or remove the wrapping `*` as necessary. You can also use this parser in other parts of the code, like when you need to hide the wrapping `*` in the filter value.

You can see that there are some existing parsers for specific filter types in `searchQueryBuilder/tokens/parsers/*` that you can reference. It might make sense to extend the string parser to look for `*`? You could also build another one if extending the old proves difficult.

Reply:
Alrighty, i've gone through and updated the two string related parsers that I found, and this seems to be functioning properly now 🥳

Reply:
There are still some funky user interactions happening that I need to peek into next 👀